packages/block-editor/src/components/inserter/menu.native.js

```diff
@@ -136,82 +134,74 @@ function InserterMenu( {
 		[ insertBlock, destinationRootClientId, insertionIndex ]
 	);
 
-	/**
-	 * Processes the inserter items to check
-	 * if there's any copied block in the clipboard
-	 * to add it as an extra item
-	 */
-	function getItems() {
-		// Filter out reusable blocks (they will be added in another tab)
-		let itemsToDisplay = items.filter(
-			( { name } ) => name !== 'core/block'
-		);
-
-		itemsToDisplay = searchItems( itemsToDisplay, filterValue );
-
-		const clipboard = getClipboard();
-		let clipboardBlock = rawHandler( { HTML: clipboard } )[ 0 ];
-
-		const canAddClipboardBlock = canInsertBlockType(
-			clipboardBlock?.name,
-			destinationRootClientId
-		);
-
-		if ( ! canAddClipboardBlock ) {
-			return itemsToDisplay;
-		}
-
-		const { icon, name } = getBlockType( clipboardBlock.name );
-		const { attributes: initialAttributes, innerBlocks } = clipboardBlock;
+	const onSelectItem = useCallback(
+		( item ) => {
+			onInsert( item );
+			onSelect( item );
+		},
+		[ onInsert, onSelect ]
+	);
 
-		clipboardBlock = {
-			id: 'clipboard',
-			name,
-			icon,
-			initialAttributes,
-			innerBlocks,
-		};
+	const onChangeSearch = useCallback(
+		( value ) => {
+			if ( ! value ) {
+				LayoutAnimation.configureNext(
+					LayoutAnimation.Presets.easeInEaseOut
+				);
+			}
+			setFilterValue( value );
+		},
+		[ setFilterValue ]
+	);
 
-		return [ clipboardBlock, ...itemsToDisplay ];
-	}
+	const showReusableBlocks = ! rootClientId;
 
 	return (
 		<BottomSheet
 			isVisible={ true }
 			onClose={ onClose }
-			hideHeader
+			onKeyboardShow={ () => setShowTabs( false ) }
+			onKeyboardHide={ () => setShowTabs( true ) }
```

Comment:
When the keyboard is shown/hide, it dispatches a layout animation and as we're also rendering different content when that happens (actually when the search form is active), these handlers are used to group all the layout changes in one single layout animation.

Reply:
Could we move these arrow functions?

Reply:
Sure! I'll create them using `useCallback` to prevent re-rendering 👍 .